### ISSUE TITLE
Patient View Pathways tab does not load

### ISSUE THREAD

**Issue by onursumer** [cBioPortal/cbioportal#8559]
Seems like Pathways tab on Patient View page is broken. Tried with many different patients. None of them loads. A couple of examples:

https://www.cbioportal.org/patient/pathways?sampleId=TCGA-AA-3561-01&studyId=coadread_tcga_pub
https://www.cbioportal.org/patient/pathways?sampleId=TCGA-06-0197-01&studyId=gbm_tcga_pub

Might be related to cBioPortal/cbioportal-frontend#3714. Need to investigate further.

**Comment by ugurdogrusoz** [comment]
This one works fine:
https://www.cbioportal.org/patient/pathways?studyId=ucec_tcga_pub&caseId=TCGA-BK-A0CC

**Comment by msalihaltun** [comment]
`PatientViewPage` component renders `PathwayMapperComponent` or `LoadingIndicator` based on the truth value of `patientViewPageStore.geneticTrackData.isComplete`. It seems right now that for some (most?) patient pages this value is never set to `true`. In the example @ugurdogrusoz linked, it is.

I don't think it is related to https://github.com/cBioPortal/cbioportal-frontend/pull/3714, I tried a local deployment using a previous version of PathwayMapper but the issue still persists.

**Comment by onursumer** [comment]
After some debugging, I found the root cause of the issue. It is related to this change:

https://github.com/cBioPortal/cbioportal-frontend/pull/3358/files#diff-b0a84573cdfa66fa186965bc88e3b13f540b646a200f79ef24ab900c00ae55bdL1411-R1550

Data within the `this.samples` does not always have `uniqueSampleKey`s. That was why we introduced `samplesWithUniqueKeys`. But seems like it has been reverted by cBioPortal/cbioportal-frontend/pull/3358. As a result, most of the times `computeGenePanelInformation` fails to return a valid value.

We need to make sure that sample data passed to `computeGenePanelInformation` has `uniqueSampleKey`s.
